### PR TITLE
Finalize create and update users API

### PR DESCRIPTION
**Pull request by blairlunceford**
## Description

Finalizes the create and update users API methods

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: workos/user_management.py [hunk 1]
```diff
@@ -160,27 +133,44 @@ def list_users(
...
             payload (dict) - The User attributes to be updated
-                user[first_name] (string) - The user's first name.
-                user[last_name] (string) - The user's last name.
-                user[email_verified] (bool) - Whether the user's email address was previously verified.
+                payload[first_name] (str) - The user's first name. (Optional)
+                payload[last_name] (str) - The user's last name. (Optional)
+                payload[email_verified] (bool) - Whether the user's email address was previously verified. (Optional)
+                payload[password] (str) - The password to set for the user. (Optional)
+                payload[password_hash] (str) - The hashed password to set for the user, used when migrating from another user store. Mutually exclusive with password. (Optional)
+                payload[password_hash_type] (str) - The algorithm originally used to hash the password, used when providing a password_hash. Valid values are bcrypt. (Optional)
```
Comment (oliverzheng) [inline workos/user_management.py:173]:
```suggestion
                payload[password_hash_type] (str) - The algorithm originally used to hash the password, used when providing a password_hash. Only valid value is 'bcrypt'. (Optional)
```

**Review by oliverzheng — APPROVED** [review]

File: workos/user_management.py [hunk 2]
```diff
@@ -160,27 +133,44 @@ def list_users(
...
-        self.request_helper.request(
-            USER_DETAIL_PATH.format(user_id),
-            method=REQUEST_METHOD_DELETE,
+        headers = {}
+
+        response = self.request_helper.request(
+            USER_PATH,
+            method=REQUEST_METHOD_POST,
+            params=user,
+            headers=headers,
```
Comment (amygdalama) [inline workos/user_management.py:156]: ooc what was this change (adding empty headers) for? or was it copy/paste?

Comment (blairlunceford) [reply workos/user_management.py:156]: This was me moving methods around (so they are in a different order), and the diff just not dealing with that well 🙃

Comment (amygdalama) [reply workos/user_management.py:156]: oooooh yeah i see my bad 😆

**Pull request merged**